roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh

```diff
@@ -5,107 +5,123 @@ set -u
 set -o pipefail
 set -x
 
-cd /storage
+echo "Installing 'unzip' ..."
+dnf install unzip -y --quiet
 
-DOWNLOAD_MAX_TIME_MIN=20
-CERT_FILE=${CERT_FILE:-}
+if [[ -z "${STORAGE_DIR:-}" ]]; then
+    STORAGE_DIR=/storage
+fi
 
-cat > checksum.sha256sum <<EOF
+cat > ${STORAGE_DIR}/checksum.sha256sum <<EOF
 113a836d90195ee1f884e704da6304dfaaecff1f023f49b6ca93c4aaae470268  annotations_trainval2017.zip
 c7908c3c9f94ba2f3340ebbeec58c25db6be8774f18d68c2f15d0e369d95baba  test2017.zip
 69a8bb58ea5f8f99d24875f21416de2e9ded3178e903f1f7603e283b9e06d929  train2017.zip
 4f7e2ccb2866ec5041993c9cf2a952bbed69647b115d0f74da7ce8f4bef82f05  val2017.zip
 e52f412dd7195ac8f98d782b44c6dd30ea10241e9f42521f67610fbe055a74f8  image_info_test2017.zip
 EOF
 
-declare -A DIR_PREFIXES=(
-    ["test2017.zip"]="zips"
-    ["train2017.zip"]="zips"
-    ["val2017.zip"]="zips"
-    ["annotations_trainval2017.zip"]="annotations"
-    ["image_info_test2017.zip"]="annotations"
-)
+CRED_FILE=${CRED_FILE:-}
+if [[ -n "${CRED_FILE}" ]]; then
+    curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+    unzip awscliv2.zip
+    ./aws/install
+    aws configure import --csv file://${CRED_FILE}
 
-if [[ -z "$CERT_FILE" ]]; then
-    echo "INFO: no cert file provided, downloading from upstream."
+    ann_check=$(ls -1q ${STORAGE_DIR}/annotations | wc -l)
+    train_check=$(ls -1q ${STORAGE_DIR}/train2017 | wc -l)
+    val_check=$(ls -1q ${STORAGE_DIR}/val2017 | wc -l)
+
+    if [[ $ann_check != 8 ]]; then
+        rm -rf ${STORAGE_DIR}/annotations
+        aws s3 cp s3://psap-coco-bucket/annotations/ ${STORAGE_DIR}/annotations --profile=cocobucket --recursive
+    fi
+    if [[ $train_check != "118287" ]]; then
```

Comment:
can you write the expected number of files in a bash array, to make it clearer what the values mean?
eg,
```
EXPECTED_NUMBER_OF_FILES=(associative array)
EXPECTED_NUMBER_OF_FILES[annotations] = 8
EXPECTED_NUMBER_OF_FILES[train2017] = 118287
...
```
you can even loop across all the keys and 
```

        rm -rf ${STORAGE_DIR}/$key
        aws s3 cp s3://psap-coco-bucket/$key/ ${STORAGE_DIR}/$key --profile=cocobucket --recursive
    fi
```
and variables must be protected by `"` as often as possible,
imagine there's a space in `STORAGE_DIR="/ stupid dirname -rf help/"`
this will execute `rm -rf / stupid dirname -rf help/`
instead of simply failing because the directory doesn't exist